crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -109,12 +110,29 @@ pub trait HandleBatchSwaps: StateWrite + Sized {
                 .into(),
         };
 
-        // Fetch the swap execution object that should have been modified during the routing and filling.
         tracing::debug!(
             ?output_data,
             ?swap_execution_1_for_2,
             ?swap_execution_2_for_1
         );
+
+        // Update the candlestick tracking
+        if let Some(se) = swap_execution_1_for_2.clone() {
+            tracing::debug!("updating candlestick for 1=>2 swap");
+            Arc::get_mut(self)
+                .expect("expected state to have no other refs")
+                .record_swap_execution(&se)
+                .await?;
+        }
+        if let Some(se) = swap_execution_2_for_1.clone() {
```

Comment:
nit: we're passing by reference below, so cloning the execution trace is unnecessary i believe